packages/app-builder-lib/src/util/appFileCopier.ts

```diff
@@ -188,32 +188,40 @@ export async function computeNodeModuleFileSets(platformPackager: PlatformPackag
   const result = new Array<ResolvedFileSet>()
   let index = 0
   const NODE_MODULES = "node_modules"
-  const getRealSource = (source: string) => {
-    const parentDir = path.dirname(source)
+  const getRealSource = (name: string, source: string) => {
+    const normalizedPath = source.split(path.sep).join("/")
+
+    if (!normalizedPath.endsWith(name)) {
+      throw new Error("Path does not end with the package name")
+    }
+    const parentDir = path.normalize(normalizedPath.slice(0, -name.length - 1))
```

Comment:
I'm not sure I understand this logic. (e.g. the `-name.length-1`)
Can you explain here or add a code comment as to what this is doing?

Reply:
Here's the English translation of the provided Chinese text:

"""
The source we currently provide is in this format: /root/path/node_modules/test or /root/path/node_modules/@electron/remote. We need to extract /root/path so that the filter in the fileMatcher can extract node_modules/test or node_modules/@electron/remote for filtering.

The issue now is with modules named like @electron/remote. The original code directly uses path.dir(path.dir) to obtain the path, which works fine for node_modules/test, but for @electron/remote, it retrieves /root/path/node_modules instead of /root/path/.

Therefore, to truncate the source using the module name, we first need to convert path.sep to / before we can perform the truncation process.
"""